scripts/create-docker-config.js

```diff
@@ -1,13 +1,13 @@
 require('dotenv').config();
-if (typeof process.env.FORCE_HTTP == 'undefined') process.env.FORCE_HTTP = 'yes';
+if (typeof process.env.FORCE_HTTP == 'undefined')
+  process.env.FORCE_HTTP = 'yes';
 const fs = require('fs').promises;
 
-(async function() {
-
+(async function () {
   // re-use mysql password
   if (!process.env.DB_PASSWORD) {
     try {
-      let current = await fs.readFile('./.env.docker');
+      let current = await fs.readFile('./.env');
```

Comment:
Het idee van dit script was: zet een env op op basis van alleen wat basis waarden in je `.env`, en verberg zo de complexiteit.

Het werkt zo nog steeds, maar omdat je hem nu overschrijft is het eenmalig. Daarna wordt je voor elke wijjziging wel geconfronteerd met die complexiteit. Dat is misschien jammer.

Kunnen we iets bedenken waarbij die beide dingen naast elkaar kunnen bestaan. Als je graag het resultaat in `.env` wilt hebben dan zou de eenvoudige versie misschien een andere naam kunnen krijgen?

Daarbij is het wellicht ook goed om te weten dat de config ook wordt gebruikt door het setup-all script, en dat wijzigingen in filenames daar ook effect kunnen hebben.

Reply:
Dit sluit beter aan op de conventie's van veel tools. Nu heb je geen aparte flag nodig die `.env.docker` inlaad, wat je vaak kunt vergeten. Als je je .env alleen gebruikte voor je overrides kun je dit dus beter omdraaien zoals het nu werkt. Bijvoorbeeld met een `.env.local` die je dan extra meegeeft aan je docker compose command of inlaad in je `docker-compose.override.yml`. 

Evt. kun je deze automatisch inladen door een export toe te voegen van de `COMPOSE_ENV_FILES` variabel: https://docs.docker.com/compose/environment-variables/envvars/#compose_env_files

```
export COMPOSE_ENV_FILES=".env,.env.local"
```

Reply:
Ja, precies, dat bedoelde ik met 'dan zou de eenvoudige versie misschien een andere naam kunnen krijgen'. Dat lijkt me werkbaar. Alleen zijn het niet zozeer local overrides, maar is het een basis voor de rest. Ik zou er dan iets van maken als .env.base, waar die config sripts dan de hele configuratie op baseren.

En dan houdt je .env.local voor local overrides.

Reply:
Wat ik juist wil vermijden is meerdere .env files waardoor je dus altijd handmatig moet meegeven dat er een bepaalde env file moet worden ingeladen. Door nu 1 `.env` file te hebben die de standaard configuratie bevat wordt deze automatisch opgepakt door de tooling en heb je geen extra flags (`--env-file`) nodig om die in te laden. Iets wat ik persoonlijk erg vervelend en fout gevoelig vind in de huidige main branch waarbij er een aparte docker-compose.yml is en een aparte .env.docker file waardoor je dus een lang command moet uitvoeren.

Op deze manier kan je er zelf voor kiezen de .env aan te passen of een eigen docker-composer.override.yml aan te maken met alleen je eigen wijzigingen. Deze wordt dan automatisch ingeladen door de tooling. Hierdoor blijft het startup command super simpel (ala `npm start`).

In je docker-compose.override.yml doe je bijvoorbeeld dit:

```yml
# docker-compose.override.yml
version: '3.4'
services:
  openstad-mysql:
    environment:
      - MYSQL_ROOT_PASSWORD="mijn eigen root password"
```

Mijn voorkeur zou zelfs uitgaan naar een `.env` in het git repo zelf met alle standaard configuratie al ingevuld waardoor je ook geen create-docker-config script meer hoeft te draaien want het werkt al out-of-the-box. Hierdoor ben je met 2 commands klaar:

```sh
$ git clone git@github.com:openstad/openstad-headless.git
$ docker-compose up
```

Maar ik snap dat hierdoor de documentatie ook herschreven moet worden op dit moment.